docs: Nginx with SSL and HTTP/2 is now default (#3970)
mattermost/docs: source/install/config-proxy-nginx.rst

2. Create a configuration file for Mattermost.

  ``sudo touch /etc/nginx/sites-available/mattermost``

On RHEL 7: ``sudo touch /etc/nginx/conf.d/mattermost``

3. Open the file ``/etc/nginx/sites-available/mattermost`` as *root* user in a text editor and replace its contents, if any, with the following lines. Make sure that you use your own values for the Mattermost server IP address and FQDN for *server_name*.

On RHEL 7, open the file ``/etc/nginx/conf.d/mattermost``.

SSL and HTTP/2 with server push are enabled in the provided configuration example.

.. note::

>   You will need get valid SSL certificates for nginx to pin the certificates properly and your browser to accept the certificate as a valid CA signed certificate. If you need an example on full configuration with pinning *Let's Encrypt*, please see `Nginx HTTP/2 & SSL full configuration guide <https://docs.mattermost.com/install/config-ssl-http2-nginx.html>`__.

Reviewer: ```suggestion
  You will need valid SSL certificates in order for NGINX to pin the certificates properly. Additionally, your browser must have permissions to accept the certificate as a valid CA signed certificate. If you need an example on full configuration with pinning Let's Encrypt, please see the `Nginx HTTP/2 & SSL full configuration guide <https://docs.mattermost.com/install/config-ssl-http2-nginx.html>`__.
```

       }
    }

    # This block is useful for debugging TLS v1.3. Please feel free to remove this

>     # and use the `$ssl_early_data` variable exposed by nginx directly should you

Reviewer: ```suggestion
    # and use the `$ssl_early_data` variable exposed by Nginx directly should you
```
